src/main/java/uk/gov/justice/hmpps/prison/service/digitalwarrant/DigitalWarrantService.java

```diff
@@ -160,7 +168,29 @@ public Integer createOffenderSentence(Long bookingId, Sentence sentence) {
         var booking = offenderBookingRepository.findByBookingId(bookingId).orElseThrow(EntityNotFoundException.withIdAndClass(bookingId, OffenderBooking.class));
         var offenderCharge = offenderChargeRepository.findById(sentence.getOffenderChargeId()).orElseThrow(EntityNotFoundException.withIdAndClass(sentence.getOffenderChargeId(), OffenderCharge.class));
         var sentenceCalcType = sentenceCalcTypeRepository.findById(new SentenceCalcType.PK(sentence.getSentenceType(), sentence.getSentenceCategory())).orElseThrow(EntityNotFoundException.withIdAndClass(sentence.getSentenceType() + " " + sentence.getSentenceCategory(), SentenceCalcType.class));
-        var courtEvent = courtCase.getCourtEvents().get(0);
+
+        var movementReason = movementReasonReferenceCodeRepository.findById(MovementReason.SENTENCING).orElseThrow(EntityNotFoundException.withIdAndClass(MovementReason.SENTENCING.getCode(), MovementReason.class));
+        var eventStatus = eventStatusReferenceCodeRepository.findById(EventStatus.COMPLETED).orElseThrow(EntityNotFoundException.withIdAndClass(EventStatus.COMPLETED.getCode(), EventStatus.class));
+        var result = offenceResultRepository.findById(OffenceResultRepository.IMPRISONMENT).orElseThrow(EntityNotFoundException.withIdAndClass(OffenceResultRepository.IMPRISONMENT, OffenceResult.class));
+        var courtEvent = CourtEvent.builder()
+            .offenderBooking(booking)
+            .courtEventType(movementReason)
+            .eventStatus(eventStatus)
+            .startTime(sentence.getSentenceDate().atTime(10, 0))
```

Comment:
10 seems a bit random 😄

Reply:
Yea! I'm finding things that are missing from the frontend so this will need to be added to the API at some point. I'm keeping track of the things missing from the first UI design.